packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m

```diff
@@ -424,7 +424,19 @@ - (void)signInWithCredential:(id)arguments
   [auth signInWithCredential:credential
                   completion:^(FIRAuthDataResult *authResult, NSError *error) {
                     if (error != nil) {
-                      result.error(nil, nil, nil, error);
+                      NSDictionary *userInfo = [error userInfo];
+                      NSError *underlyingError = [userInfo objectForKey:NSUnderlyingErrorKey];
+
+                      NSDictionary *firebaseDictionary =
+                          underlyingError.userInfo[@"FIRAuthErrorUserInfoDeserializedResponseKey"];
+
+                      if (firebaseDictionary != nil && firebaseDictionary[@"message"] != nil) {
+                        // error from firebase-ios-sdk is buried in underlying error.
+                        NSString *code = [NSString stringWithFormat:@"%li", underlyingError.code];
+                        result.error(code, firebaseDictionary[@"message"], nil, nil);
```

Comment:
I'd probably stick with the `nil` error code since it defaults to `internal-error` - this change above changes the code to `3` which isn't helpful

TLDR; lets only update the message